tracer/test/Datadog.Trace.Tools.Runner.IntegrationTests/GacCommandTests.cs

```diff
@@ -96,11 +99,94 @@ public void GacCompleteTest()
 
             // it shouldn't be in the gac
             result.Should().NotBe(0);
-            consoleGet3.Output.Should().Contain($"Error getting '{assemblyName}' from the GAC");
+            consoleGet3.Output.Should().Contain($"Error getting '{DummyAssemblyName}' from the GAC");
+        }
+    }
+
+    [SkippableFact]
+    [Trait("RunOnWindows", "True")]
+    public void GacWithMultipleVersions()
+    {
+        Skip.IfNot(FrameworkDescription.Instance.IsWindows());
+
+        // We skip the test if we don't have permissions
+        Skip.IfNot(Gac.AdministratorHelper.IsElevated);
+
+        const string DummyAssemblyName = "DummyLibrary";
+        var rootFolder = Path.GetDirectoryName(typeof(GacCommandTests).Assembly.Location)!;
+        var assemblyPathV1 = Path.GetFullPath(Path.Combine(rootFolder, $"{DummyAssemblyName}V1.dll"));
+        var assemblyPathV2 = Path.GetFullPath(Path.Combine(rootFolder, $"{DummyAssemblyName}V2.dll"));
+        var assemblyPathTarget = Path.GetFullPath(Path.Combine(rootFolder, $"{DummyAssemblyName}.dll"));
+
+        // GET
+        using (var consoleGet = ConsoleHelper.Redirect())
+        {
+            var result = Program.Main($"gac get {DummyAssemblyName}".Split(' '));
+
+            // it shouldn't be in the gac
+            result.Should().NotBe(0);
+            consoleGet.Output.Should().Contain($"Error getting '{DummyAssemblyName}' from the GAC");
+        }
+
+        // INSTALL
+
+        // install the first one
+        using (var consoleInstall = ConsoleHelper.Redirect())
+        {
+            File.Copy(assemblyPathV1, assemblyPathTarget, true);
+            var result = Program.Main($"gac install {assemblyPathTarget}".Split(' '));
+
+            // should be successful
+            result.Should().Be(0);
+        }
+
+        // install the second one
+        using (var consoleInstall = ConsoleHelper.Redirect())
+        {
+            File.Copy(assemblyPathV2, assemblyPathTarget, true);
+            var result = Program.Main($"gac install {assemblyPathTarget}".Split(' '));
+
+            // should be successful
+            result.Should().Be(0);
+        }
+
+        // GET
+        using (var consoleGet2 = ConsoleHelper.Redirect())
+        {
+            var result = Program.Main($"gac get {DummyAssemblyName}".Split(' '));
+
+            // should be in the gac
+            result.Should().Be(0);
+            consoleGet2.Output.Should().Contain("Assembly Found!");
+
+            // Check installation of version 1
+            consoleGet2.Output.Should().Contain($"- FullName={DummyAssemblyName}, Version=1.0.0.0, Culture=neutral, PublicKeyToken=fb8e173370c0fc9d");
+
+            // Check installation of version 2
+            consoleGet2.Output.Should().Contain($"- FullName={DummyAssemblyName}, Version=2.0.0.0, Culture=neutral, PublicKeyToken=fb8e173370c0fc9d");
+        }
+
+        // UNINSTALL
+        using (var consoleUninstall = ConsoleHelper.Redirect())
```

Comment:
We should probably uninstall the second one afterwards too?

Reply:
I cannot specify the version to uninstall so currently the Uninstall option uninstall all versions. did you have something in FleetInstaller?

Reply:
Ah, it's maybe because we're passing in a filepath identifier when we register in the GAC, and we can use that when we uninstall to identify the correct version 🤔